packages/swapper/src/swappers/ButterSwap/swapperApi/checkTradeStatus.ts

```diff
@@ -13,24 +15,68 @@ const BUTTER_SWAP_STATES = (() => ({
   Failed: 6,
 }))()
 
+// Cache relay IDs to avoid repeated API calls during polling
+// We use a module-level cache instead of storing in swap.metadata because:
+// 1. The relay ID is only available after transaction submission (not at quote time like chainflipSwapId)
+// 2. checkTradeStatus doesn't have a mechanism to update swap state during polling
+// 3. This approach is simpler and doesn't require changes to core types or polling logic
+const relayIdCache = new Map<string, number>()
+
 export const checkTradeStatus = async (input: CheckTradeStatusInput): Promise<TradeStatus> => {
   const { txHash } = input
   try {
-    const infoResult = await getBridgeInfoBySourceHash(txHash)
-    if (infoResult.isErr()) {
+    let relayId = relayIdCache.get(txHash)
+
+    // Only fetch relay info by source hash if we don't have the relay ID cached
+    if (!relayId) {
+      const infoResult = await getBridgeInfoBySourceHash(txHash)
+      if (infoResult.isErr()) {
+        return {
+          status: TxStatus.Unknown,
+          buyTxHash: undefined,
+          message: infoResult.unwrapErr().message,
+        }
+      }
+      const basicInfo = infoResult.unwrap()
+      if (!basicInfo || !basicInfo.id) {
+        return { status: TxStatus.Unknown, buyTxHash: undefined, message: undefined }
+      }
+
+      // Cache the relay ID for future calls
+      relayId = basicInfo.id
+      relayIdCache.set(txHash, relayId)
+    }
+
+    // Get the detailed relay info using the cached ID
+    const detailedInfoResult = await getBridgeInfoById(relayId)
+    if (detailedInfoResult.isErr()) {
       return {
         status: TxStatus.Unknown,
         buyTxHash: undefined,
-        message: infoResult.unwrapErr().message,
+        message: detailedInfoResult.unwrapErr().message,
       }
     }
-    const info = infoResult.unwrap()
-    if (!info) {
+    const detailedInfo = detailedInfoResult.unwrap()
+    if (!detailedInfo) {
       return { status: TxStatus.Unknown, buyTxHash: undefined, message: undefined }
     }
+
+    // Use toHash as the destination chain tx hash if present
+    const buyTxHash = detailedInfo.toHash ?? undefined
+    const relayerTxHash = detailedInfo.relayerHash ?? undefined
+    const relayerExplorerTxLink = detailedInfo.relayerChain?.scanUrl ?? undefined
+
     // State mapping: 0 = pending, 1 = complete, 6 = refunded/failed
     let status: TxStatus = (() => {
-      switch (info.state) {
+      // For BTC, be optimistic on buyTxHash as we update the balance once the TX hits the mempool
+      if (
+        buyTxHash &&
+        butterSwapChainIdToChainId(Number(detailedInfo.toChain.chainId)) ===
+          KnownChainIds.BitcoinMainnet
+      )
+        return TxStatus.Confirmed
```

Comment:
Don't think we need to keep this BTC specific, if a destination Tx was broadcasted, then the swap is complete, regardless of destination chain

Reply:
What if the TX fails/reverts etc though?

Reply:
If there's a revert there's no Txid since there's no Tx, and EVM Tx failures are so rare, literally barely ever saw one anywhere ever, but assuming that was a thing, that wouldn't be the responsibility of this method 🙏🏽  since as far as the *swap* is concerned, it *is* complete!

Reply:
Works for me: https://github.com/shapeshift/web/pull/9833/commits/54b3d497439df1096a743f573c57542ca6509e7b